client/express-checkout/event-handlers.js

```diff
@@ -99,3 +100,61 @@ export const onConfirmHandler = async (
 		return abortPayment( event, e.message );
 	}
 };
+
+/**
+ * Generates a Pay for Order handler based on a particular order.
+ * Contains pretty much the same logic in `onConfirmHandler`.
+ *
+ * @param {integer} order The ID of the order that is being paid.
+ * @return {Function} The handler.
+ */
+export const payForOrderHandler = ( order ) => async (
```

Comment:
Nit: This seems to be a duplicate of `onConfirmHandler()`, are we able to remove some of the duplication by using a helper function? Or by passing an `option.<pay_for_order>` that allows for an if-else statement that switches between pay-for-order and create-order.

Although not necessary, it would be nice so that we don't have to maintain two functions.

Reply:
> it would be nice so that we don't have to maintain two functions

Totally agree with that 💯 I refactored this in [fa4559c](https://github.com/Automattic/woocommerce-payments/pull/8994/commits/fa4559c425d6ed640b67b658cfd96ee7e1ec14e8) and the respective tests in [94bef90](https://github.com/Automattic/woocommerce-payments/pull/8994/commits/94bef90c558fae55132f3d32565b8b7a8a712d3d) 👍